flexget/components/notify/notifiers/slack.py

```diff
@@ -15,25 +15,86 @@
+                version: 1
                 web_hook_url: <string>
                 [channel: <string>] (override channel, use "@username" or "#channel")
                 [username: <string>] (override username)
                 [icon_emoji: <string>] (override emoji icon)
                 [icon_url: <string>] (override emoji icon)
                 [attachments: <array>[<object>]] (override attachments)
 
+    Example (version 2):
+
+      notify:
+        entries:
+          via:
+            - slack2:
+                version: 2
+                web_hook_url: "https://hooks.slack.com/services/..."
+                blocks:
+                  - section:
+                      text: '<{{ tvdb_url }}|{{ series_name }} ({{ series_id }})>'
+                  - section:
+                      text: '{{ tvdb_ep_overview }}'
+                      image:
+                        url: "https://api.slack.com/img/blocks/bkb_template_images/plants.png"
+                        alt_text: 'plants'
+                      fields:
+                        - '*Score*'
+                        - '*Genres*'
+                        - '{{ imdb_score }}'
+                        - "{{ imdb_genres | join(', ') | title }}"
+                        - '*Rating*'
+                        - '*Cast*'
+                        - '{{ imdb_mpaa_rating }}'
+                        - >
+                          {% for key, value in imdb_actors.items() %}{{ value }}{% if not loop.last %}, {% endif %}
+                          {% endfor %}
+                  - image:
+                      url: "{{ tvdb_banner }}"
+                      alt_text: "{{ series_name }} Banner"
+                  - context:
+                      - image:
+                          url: 'https://image.freepik.com/free-photo/red-drawing-pin_1156-445.jpg'
+                          alt_text: 'red pin'
+                      - text: ':round_pushpin:'
+                      - text: 'Task: {{ task }}'
     """
 
     schema = {
+        'definitions': {
+            'image_block': {
+                'type': 'object',
+                'properties': {
+                    'url': {'type': 'string', 'format': 'uri'},
+                    'alt_text': {'type': 'string'},
+                },
+                'required': ['url', 'alt_text'],
+                'additionalProperties': False,
+            },
+            'image_block_w_title': {
+                'type': 'object',
+                'properties': {
+                    'url': {'type': 'string', 'format': 'uri'},
+                    'title': {'type': 'string'},
+                    'alt_text': {'type': 'string'},
+                },
+                'required': ['url', 'alt_text'],
+                'additionalProperties': False,
+            }
+
+        },
+
         'type': 'object',
         'properties': {
             'web_hook_url': {'type': 'string', 'format': 'uri'},
+            'version': {'type': 'integer', 'enum': [1, 2], 'default': 1},
```

Comment:
interesting. Perhaps a simpler approach would be to declare the two schema independently and then use `oneOf` between them. WDYT?

Reply:
Yeah, we don't generally do versioning on configuration. Version can be inferred from using "blocks" key here?